index.js

```diff
@@ -0,0 +1,41 @@
+const timeouts = {},
+  keys = document.querySelector('.keys')
+document.addEventListener('DOMContentLoaded', () => {
+  window.addEventListener('keydown', (e) => {
+    const keyCode = e.keyCode
+    const key = document.querySelector(`button[data-key="${keyCode}"]`)
+    if (key) {
+      const audio = document.querySelector(`audio[data-key="${keyCode}"]`)
+      if (timeouts[keyCode]) clearTimeout(timeouts[keyCode])
+      audio.play()
+      audio.currentTime = 0
+      const clone = key.cloneNode(true)
+      clone.classList.add('active')
+      timeouts[keyCode] = setTimeout(() => {
+        clone.classList.remove('active')
+      }, 500)
+      key.parentElement.replaceChild(clone, key)
+    }
+  })
+
+  keys.addEventListener('click', (e) => {
+    let element = e.target
+    while (element && element.tagName !== 'BUTTON') {
+      element = element.parentElement
+    }
+
+    if (element) {
```

Comment:
Better to use `if (!element) return;` just in the beginning.  It is is early exit from function.